.azure-pipelines.yml

```diff
@@ -0,0 +1,248 @@
+        rustup update $RUSTUP_TOOLCHAIN
+        rustup default $RUSTUP_TOOLCHAIN
+        rustc --version
+      displayName: Install rust (*nix)
+      condition: ne( variables['Agent.OS'], 'Windows_NT' )
+    - script: |
+        rustup update %RUSTUP_TOOLCHAIN%
+        rustup default %RUSTUP_TOOLCHAIN%
+        rustc --version
+      displayName: Install rust (Win)
+      condition: eq( variables['Agent.OS'], 'Windows_NT' )
+    - script: |
+        rustup component add rustfmt
+      displayName: Add rustfmt
+      condition: and(eq(variables['rustup_toolchain'], 'stable'), ne( variables['build_type'], 'release' ),
+                     eq(variables['Agent.OS'], 'Darwin'))
+    - script: |
+        powershell -Command "$ProgressPreference = 'SilentlyContinue'; iwr -outf %TEMP%\LLVM-8.0.0-win64.exe https://rust-lang-ci2.s3.amazonaws.com/rust-ci-mirror/LLVM-8.0.0-win64.exe"
+        %TEMP%\LLVM-8.0.0-win64.exe /S /NCRC
+      condition: eq(variables['Agent.OS'], 'Windows_NT')
+      displayName: Install clang (Windows)
+    - bash: echo "##vso[task.setvariable variable=RUSTC_VERSION;]`rustc --version`"
+      displayName: Set rustc version string for caching
+    - bash: |
+        echo "##vso[task.setvariable variable=tagName;]`echo $BUILD_SOURCEBRANCH | sed -e 's|refs/tags/||'`"
+      displayName: Set tag name
+      condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/')
+    - bash: |
+        echo "##vso[task.setvariable variable=tagName;]latest-master"
+      displayName: Set tag name to "latest-master"
+      condition: eq(variables['Build.SourceBranch'], 'refs/heads/master')
+
+    - bash: |
+        set -e
+        curl -fo /usr/local/bin/sccache https://rust-lang-ci2.s3.amazonaws.com/rust-ci-mirror/2018-04-02-sccache-x86_64-apple-darwin
+        chmod +x /usr/local/bin/sccache
+      displayName: Install sccache (OSX)
+      condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+
+    - bash: |
+        set -e
+        sudo apt-get install sccache -y
+      displayName: Install sccache (Linux)
+      condition: and(succeeded(), eq(variables['Agent.OS'], 'Linux'))
+
+    - script: |
+        md sccache
+        powershell -Command "$ProgressPreference = 'SilentlyContinue'; iwr -outf sccache\sccache.exe https://rust-lang-ci2.s3.amazonaws.com/rust-ci-mirror/2018-04-26-sccache-x86_64-pc-windows-msvc"
+        echo ##vso[task.prependpath]%CD%\sccache
+      displayName: Install sccache (Windows)
+      condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'))
+
+    - bash: |
+        mkdir $SCCACHE_DIR
+        touch $SCCACHE_DIR/foo
+        sccache --show-stats
+
+    - script: cargo fetch
+      displayName: Fetch cargo dependencies
+    - script: cargo fmt --all -- --check
+      displayName: Check formatting
+      condition: and(eq(variables['rustup_toolchain'], 'stable'), ne( variables['build_type'], 'release' ),
+                     eq(variables['Agent.OS'], 'Darwin'))
```

Comment:
In terms of `condition` weridness, this is where I find having a [separate `rustfmt` job](https://github.com/rust-lang/git2-rs/blob/1d33b290e034d0d85ac0e4bc0ac64cebfa8cfbc7/azure-pipelines.yml#L5-L16) to be pretty clean